aiven/resource_flink_table.go

```diff
@@ -0,0 +1,219 @@
+package aiven
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var aivenFlinkTableSchema = map[string]*schema.Schema{
+	"project": {
+		Type:        schema.TypeString,
+		Description: "Project to link the Flink Table to",
+		Required:    true,
+		ForceNew:    true,
+	},
+	"service_name": {
+		Type:        schema.TypeString,
+		Description: "Service to link the Flink Table to",
+		Required:    true,
+		ForceNew:    true,
+	},
+	"table_id": {
+		Type:        schema.TypeString,
+		Description: "Table id of the flink table",
+		Computed:    true,
+		ForceNew:    true,
+	},
+	"table_name": {
+		Type:        schema.TypeString,
+		Description: "Table name of the flink table",
+		Required:    true,
+		ForceNew:    true,
+	},
+	"integration_id": {
+		Type:        schema.TypeString,
+		Description: "Id of the service integration this table applies to",
+		Required:    true,
+		ForceNew:    true,
+	},
+	"jdbc_table": {
+		Type:        schema.TypeString,
+		Description: "Table name in Database, used as a source/sink. (PG integration only)",
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"kafka_topic": {
+		Type:        schema.TypeString,
+		Description: "Topic name, used as a source/sink. (Kafka integration only)",
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"like_options": {
+		Type:        schema.TypeString,
+		Description: "Clause can be used to create a table based on a definition of an existing table",
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"partitioned_by": {
+		Type:        schema.TypeString,
+		Description: "A column from a schema, table will be partitioned by",
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"schema_sql": {
+		Type:        schema.TypeString,
+		Description: "Source/Sink table schema",
+		Required:    true,
+		ForceNew:    true,
+	},
+}
+
+func resourceFlinkTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceFlinkTableCreate,
+		ReadContext:   resourceFlinkTableRead,
+		DeleteContext: resourceFlinkTableDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceFlinkTableState,
+		},
+
+		Schema: aivenFlinkTableSchema,
+	}
+}
+
+func resourceFlinkTableRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project, serviceName, tableId := splitResourceID3(d.Id())
+
+	r, err := client.FlinkTables.Get(project, serviceName, aiven.GetFlinkTableRequest{TableId: tableId})
+	if err != nil {
+		return diag.FromErr(resourceReadHandleNotFound(err, d))
+	}
+
+	if err := d.Set("project", project); err != nil {
```

Comment:
Let's set here all the schema fields that we can, applicable for `import` diff.

Reply:
i removed the import function since many fields cannot be imported at all; the GET response has too little information, otherwise i removed all SET calls from create